tests/test_blscache.py

```diff
@@ -0,0 +1,164 @@
+    assert cached_bls_old.aggregate_verify(
+        pks_half_bytes, msgs_half, agg_sig_half, True
+    )
+
+    # Verify with partial cache hit
+    assert cached_bls.aggregate_verify(pks, msgs, agg_sig)
+    assert cached_bls_old.aggregate_verify(pks_bytes, msgs, agg_sig, True)
+
+    # Verify with full cache hit
+    assert cached_bls.aggregate_verify(pks, msgs, agg_sig)
+    assert cached_bls_old.aggregate_verify(pks_bytes, msgs, agg_sig)
+
+    # Use a small cache which can not accommodate all pairings
+    bls_cache = BLSCache(n_keys // 2)
+    local_cache = LRUCache(n_keys // 2)
+    # Verify signatures and cache pairings one at a time
+    for pk, msg, sig in zip(pks_half, msgs_half, sigs_half):
+        assert bls_cache.aggregate_verify([pk], [msg], sig)
+        assert cached_bls_old.aggregate_verify(
+            [bytes(pk)], [msg], sig, True, local_cache
+        )
+
+    # Verify the same messages with aggregated signature (full cache hit)
+    assert bls_cache.aggregate_verify(pks_half, msgs_half, agg_sig_half)
+    assert cached_bls_old.aggregate_verify(
+        pks_half_bytes, msgs_half, agg_sig_half, False, local_cache
+    )
+
+    # Verify more messages (partial cache hit)
+    assert bls_cache.aggregate_verify(pks, msgs, agg_sig)
+    assert cached_bls_old.aggregate_verify(pks_bytes, msgs, agg_sig, False, local_cache)
+
+
+def test_cached_bls_repeat_pk():
+    cached_bls = BLSCache()
+    n_keys = 400
+    seed = b"a" * 32
+    sks = [AugSchemeMPL.key_gen(seed) for i in range(n_keys)] + [
+        AugSchemeMPL.key_gen(std_hash(seed))
+    ]
+    pks = [sk.get_g1() for sk in sks]
+    pks_bytes = [bytes(sk.get_g1()) for sk in sks]
+
+    msgs = [("msg-%d" % (i,)).encode() for i in range(n_keys + 1)]
+    sigs = [AugSchemeMPL.sign(sk, msg) for sk, msg in zip(sks, msgs)]
+    agg_sig = AugSchemeMPL.aggregate(sigs)
+
+    assert AugSchemeMPL.aggregate_verify([pk for pk in pks], msgs, agg_sig)
+
+    assert cached_bls.aggregate_verify(pks, msgs, agg_sig)
+    assert cached_bls_old.aggregate_verify(pks_bytes, msgs, agg_sig, True)
+
+
+def test_empty_sig():
+    sig = AugSchemeMPL.aggregate([])
+    cached_bls = BLSCache()
+    assert cached_bls.aggregate_verify([], [], sig)
+    assert cached_bls_old.aggregate_verify([], [], sig)
+
+
+def test_bad_cache_size():
+    with pytest.raises(ValueError) as exc_info:
+        bls_cache = BLSCache(0)
+
+    assert str(exc_info.value) == "Cannot have a cache size less than one."
+
+    with pytest.raises(OverflowError) as exc_info:
+        bls_cache = BLSCache(-1)
+
+    assert str(exc_info.value) == "can't convert negative int to unsigned"
+
+    with pytest.raises(OverflowError) as exc_info:
+        bls_cache = BLSCache(-100000)
+
+    assert str(exc_info.value) == "can't convert negative int to unsigned"
+
+    with pytest.raises(OverflowError) as exc_info:
+        bls_cache = BLSCache(-9223372036854775809)
+
+    assert str(exc_info.value) == "can't convert negative int to unsigned"
```

Comment:
instead of these checks, you can add another parameter to `pytest.raises()`, `match="can't convert negative int to unsigned"`